mappings/net/minecraft/client/render/debug/StructureDebugRenderer.mapping

```diff
@@ -1 +1,12 @@
 CLASS net/minecraft/class_870 net/minecraft/client/render/debug/StructureDebugRenderer
+	FIELD field_4624 client Lnet/minecraft/class_310;
+	FIELD field_4625 alliances Ljava/util/Map;
```

Comment:
"Alliances"? This determines whether the box is rendered in green or blue. How can you possibly get "alliances" based on just that info.

Reply:
green = friends
blue = enemies

Reply:
Firstly, I'd say red = enemies.

Secondly:
 - green = enabled, blue = disabled
 - green = fully generated, blue = not fully generated
 - green = absolutely anything, blue = absolutely anything else

Don't make bad guesses or look at the Mojang mappings if you don't know what something is.  This along with other fields whose meaning is unknown should stay unmapped.

Reply:
Will you kindly fuck off with the "Mojang mappings" bs.

Basically all the names that were present here were already Mojang names. Bad guesses I will accept, but I'm not the one who's been using Mojang names.

After checking the code, I remember that I was thinking of alliances as "villagers" vs "pillagers", or "friendly to player" and "unfriendly to player". I remembered them being a different colour in 1.14 but didn't see where the colour parameters were in 1.15.

Now that I actually know that how the rendering works, I can see that `true = green` and `false = blue`.

I still believe it's representative of some kind of alliance, because I can't see them using blue/green for generated/ungenerated. I don't think you would even be able to see ungenerated structures, because they'd be generated as soon as they get close enough for you to see them.

My thought process is that they wanted to find the pillager outposts and villagers (for debugging) and may have flagged strongholds as well, though that's just a theory.